app/src/main/java/com/asafvaron/themoviedbtest/adapters/MoviesGridAdapter.java

```diff
@@ -0,0 +1,110 @@
+package com.asafvaron.themoviedbtest.adapters;
+
+import android.content.Context;
+import android.graphics.Rect;
+import android.support.annotation.DimenRes;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.asafvaron.themoviedbtest.MyApp;
+import com.asafvaron.themoviedbtest.R;
+import com.asafvaron.themoviedbtest.objects.MyMovie;
+import com.bumptech.glide.Glide;
+
+import java.util.List;
+
+/**
+ * Created by asafvaron on 19/02/2017.
+ */
+public class MoviesGridAdapter extends RecyclerView.Adapter<MoviesGridAdapter.MovieViewHolder> {
+    private static final String TAG = MoviesGridAdapter.class.getSimpleName();
+    private final Context mContext;
+    private final RecyclerView mRvGridList;
+    private final List<MyMovie> mData;
+    private final LayoutInflater mInflater;
+
+    public MoviesGridAdapter(Context context, RecyclerView rv, List<MyMovie> data) {
+        this.mContext = context;
+        this.mRvGridList = rv;
+        this.mData = data;
+        this.mInflater = LayoutInflater.from(mContext);
+        init();
+    }
+
+    private void init() {
+        mRvGridList.setHasFixedSize(true);
+        mRvGridList.setLayoutManager(new GridLayoutManager(mContext, 2));
+        mRvGridList.setItemAnimator(new DefaultItemAnimator());
+        // set itemDecoration for grid spacing
+        GridSpacingItemDecoration decoration = new GridSpacingItemDecoration(mContext, R.dimen.grid_item_offset);
+        mRvGridList.addItemDecoration(decoration);
+        mRvGridList.setAdapter(this);
+        this.notifyDataSetChanged();
```

Comment:
bad practice... 
When using recyclerview you should almost never use notifyDataSetChanged. 
Also, Why hold all the references inside the adapter?